model_compression_toolkit/qat/pytorch/quantizer/ste_rounding/symmetric_ste.py

```diff
@@ -108,4 +107,55 @@ def __call__(self,
         w0 = ste_round(inputs / self.delta_tensor)
         w1 = ste_clip(w0, min_val=self.min_int, max_val=self.max_int)
         w_q = self.delta_tensor * w1
-        return w_q
\ No newline at end of file
+        return w_q
+
+
+class STEActivationQuantizer(qi.BasePytorchQuantizer):
+    """
+    Trainable constrained quantizer to quantize a layer activations.
+    """
+
+    def __init__(self, quantization_config: NodeActivationQuantizationConfig):
+        """
+        Initialize a STEActivationQuantizer object with parameters to use
+        for symmetric or power of two quantization.
+
+        Args:
+            quantization_config: node quantization config class
+        """
+        super().__init__(quantization_config,
+                         qi.QuantizationTarget.Activation,
+                         [qi.QuantizationMethod.POWER_OF_TWO, qi.QuantizationMethod.SYMMETRIC])
+        self.power_of_two = quantization_config.activation_quantization_method == QuantizationMethod.POWER_OF_TWO
+        self.sign = quantization_config.activation_quantization_params['is_signed']
+        np_threshold_values = quantization_config.activation_quantization_params[C.THRESHOLD]
+        self.threshold_tensor = torch.Tensor([np_threshold_values])
+        self.num_bits = quantization_config.activation_n_bits
+        self.quantizer_parameters = {}
+
+    def initialize(self):
+        """
+        Add threshold variables to layer.
+        """
+        ptq_threshold_tensor = nn.Parameter(self.threshold_tensor, requires_grad=True)
```

Comment:
I think on first phase the activation quantizer shouldn't be trained

Reply:
Don't see what's wrong if it does as it's supposed to be.